beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -191,6 +193,94 @@ func (s *Server) SubmitAggregateAndProofs(w http.ResponseWriter, r *http.Request
 	}
 }
 
+// SubmitAggregateAndProofsV2 verifies given aggregate and proofs and publishes them on appropriate gossipsub topic.
+func (s *Server) SubmitAggregateAndProofsV2(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.SubmitAggregateAndProofs")
+	defer span.End()
+
+	var req structs.SubmitAggregateAndProofsRequestV2
+	err := json.NewDecoder(r.Body).Decode(&req.Data)
```

Comment:
i'm not confident this works, shouldn't this panic? req is nil

Reply:
good question, I'm surprised too

Reply:
good question, I'm surprised too

Reply:
From what I have understood it only works because `Data` is a slice. The JSON decoder can apparently handle nil slices when decoding and will allocate new slices when needed